packages/beacon-node/src/chain/blocks/types.ts

```diff
@@ -21,38 +21,105 @@ export enum BlockSource {
   byRoot = "req_resp_by_root",
 }
 
+export enum GossipedInputType {
+  block = "block",
+  blob = "blob",
+  dataColumn = "data_column",
+}
+
+interface CachedDataItem {
+  cacheId: number;
+}
+type Availability<T> = {
+  availabilityPromise: Promise<T>;
+  resolveAvailability: (data: T) => void;
+};
+
+/**
+ *
+ * Deneb Blob Format Types
+ *
+ */
 /** Enum to represent where blobs come from */
 export enum BlobsSource {
   gossip = "gossip",
   api = "api",
   byRange = "req_resp_by_range",
   byRoot = "req_resp_by_root",
 }
+type ForkBlobsInfo = {
+  fork: ForkPostDeneb & ForkPreFulu;
+};
+export type BlockInputBlobs = ForkBlobsInfo & {
+  blobs: deneb.BlobSidecars;
+  blobsSource: BlobsSource;
+};
+export type BlobsCacheMap = Map<number, deneb.BlobSidecar>;
+export type CachedBlobs = CachedDataItem &
+  ForkBlobsInfo &
+  Availability<BlockInputBlobs> & {
+    blobsCache: BlobsCacheMap;
+  };
 
-export enum GossipedInputType {
-  block = "block",
-  blob = "blob",
+/**
+ *
+ * PeerDAS Column Format Types
+ *
+ */
+export enum DataColumnsSource {
```

Comment:
seems like this is meant as a separate comment, not jsdoc for `DataColumnsSource` enum
```suggestion
 */
 
export enum DataColumnsSource {
```

Reply:
updated in #8161